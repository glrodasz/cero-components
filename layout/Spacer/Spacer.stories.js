import { Spacer, options, styles } from '.'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Spacer, styles)
const ListTemplate = getListTemplate(Spacer, styles)

export default {
  title: 'Layout/Spacer/Default',
  component: Spacer,
  args: {
    isVisible: true,
  },
  argTypes: {
    size: getOptionsArgTypes(options.sizes),
  },
  parameters: { __sb: { fd: 'row' } },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
