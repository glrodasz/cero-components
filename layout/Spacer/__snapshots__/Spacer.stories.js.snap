// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Spacer/Default Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="spacer is-visible"
    style={
      Object {
        "display": "inline-block",
        "height": 0,
        "width": 0,
      }
    }
  />
</div>
`;

exports[`Storyshots Layout/Spacer/Default Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="spacer is-visible"
    style={
      Object {
        "display": "inline-block",
        "height": 0,
        "width": 0,
      }
    }
  />
  <div
    className="spacer is-visible"
    style={
      Object {
        "display": "inline-block",
        "height": 5,
        "width": 5,
      }
    }
  />
  <div
    className="spacer is-visible"
    style={
      Object {
        "display": "inline-block",
        "height": 15,
        "width": 15,
      }
    }
  />
  <div
    className="spacer is-visible"
    style={
      Object {
        "display": "inline-block",
        "height": 20,
        "width": 20,
      }
    }
  />
  <div
    className="spacer is-visible"
    style={
      Object {
        "display": "inline-block",
        "height": 40,
        "width": 40,
      }
    }
  />
</div>
`;
